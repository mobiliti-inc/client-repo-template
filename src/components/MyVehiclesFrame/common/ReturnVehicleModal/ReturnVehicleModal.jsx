/* eslint-disable camelcase */// API returns lots of snake_case data
import moment from 'moment';

import Modal from '../../../Modal/Modal';
import CustomDatePicker from '../../../CustomDatePicker/CustomDatePicker';
import styles from './ReturnVehicleModal.scss';
import { iconCalendar, iconTime } from '../../../../assets/icons';
import Button from '../../../Button/Button';
import DropDown from '../../../Dropdown/Dropdown';

class ReturnVehicleModal extends React.PureComponent {
	static propTypes = {
		isReturnScheduleLoading: PropTypes.bool.isRequired,
		onSelectDropOffDay: PropTypes.func.isRequired,
		onSelectDropOffTime: PropTypes.func.isRequired,
		onScheduleVehicleReturn: PropTypes.func.isRequired,
		selectedDropOffTime: PropTypes.string.isRequired,
		selectedDropOffDay: PropTypes.string.isRequired,
		subscriptionID: PropTypes.string.isRequired,
		closeModal: PropTypes.func.isRequired,
		scrolling: PropTypes.bool.isRequired,
		subscriptionEndDate: PropTypes.string.isRequired,
		modalVisible: PropTypes.bool.isRequired,
		dealerOpenDays: PropTypes.arrayOf(PropTypes.shape({
			open_day: PropTypes.string,
			hours: PropTypes.arrayOf(PropTypes.string)
		})).isRequired
	}
	state = {
		hours: [''],
		daySelected: false
	}

	onDateChange = (date) => {
		const formattedDate = moment(date.value, 'MM/DD/YYYY').format('M/D/YYYY');
		const { dealerOpenDays } = this.props;
		return dealerOpenDays.forEach((day) => {
			const { open_day } = day;
			if (formattedDate === open_day) {
				this.props.onSelectDropOffDay(open_day);
				this.props.onSelectDropOffTime(day.hours[0]);
				this.setState({
					hours: day.hours,
					daySelected: true,
					vehicleID: this.props.vehicleID,
				});
			}
		});
	}

	onTimeChange = async time => {
		const selectedTime = await this.reverseFormatTime(time);
		if (selectedTime) {
			await this.props.onSelectDropOffTime(selectedTime);
			await this.formatHours(this.state.hours);
		}
	}

	formatHours = (hours) => hours.map((hour, index) => ({
		value: this.formatTime(hour),
		current: this.props.selectedDropOffTime ?
			hour === this.props.selectedDropOffTime : index === 0
	}));

	formatTime = (time) => time.replace(/:00/g, '');

	reverseFormatTime = async time => {
		let selectedHour;
		this.state.hours.forEach(hour => {
			if (time === this.formatTime(hour)) {
				selectedHour = hour;
			}
		});
		return selectedHour;
	};

	isDayBlocked = (day) => {
		const { dealerOpenDays } = this.props;
		const lastDay = dealerOpenDays.length - 1;
		return day.format('dddd') === 'Sunday' || day.isBefore(dealerOpenDays[0].open_day) ||
		day.isAfter(dealerOpenDays[lastDay].open_day);
	}

	scheduleReturn = () => {
		const { selectedDropOffTime, selectedDropOffDay, subscriptionID } = this.props;
		const data = {
			dateAndTime: {
				return_date: selectedDropOffDay,
				return_time: selectedDropOffTime
			},
			subscriptionID
		};
		this.props.onScheduleVehicleReturn(data);
	}

	disableButton = () => !this.props.selectedDropOffDay || !this.props.selectedDropOffTime ||
	!this.state.daySelected || this.state.vehicleID !== this.props.vehicleID;

	render() {
		const {
			modalVisible,
			scrolling,
			subscriptionEndDate,
			closeModal,
			selectedDropOffDay,
		} = this.props;

		return (
			<Modal
				plain
				visible={modalVisible}
				overlay="dark-gray"
				onIconClose={closeModal}
				scrolling={scrolling}
				modalParentClass={styles.modal}
			>
				<p styleName="title">Schedule Vehicle Return</p>
				<p styleName="sub-title">If you are returning your vehicle after one month of use you could receive a credit based
					on your vehicle usage for the current month.
				</p>
				<div styleName="subscription-end-date">
					<p>{moment(subscriptionEndDate, 'MM/DD/YYYY').format('dddd, MMMM Do')}</p>
					<p>Subscription End Date</p>
				</div>
				<p styleName="drop-off-date-title">Drop off Date/Time</p>
				<div styleName="date-selection-ctn">
					<div styleName="date-picker-ctn">
						<img src={iconCalendar} alt="calendar-icon" />
						<CustomDatePicker
							placeholder="Date"
							isOutsideRange={this.isDayBlocked}
							onDatePicked={this.onDateChange}
							numberOfYears={2}
						/>
					</div>
					{ selectedDropOffDay && this.state.daySelected &&
					<div styleName="time-drop-down-ctn">
						<img src={iconTime} alt="time-icon" />
						<DropDown
							header="Select a Time"
							data={this.formatHours(this.state.hours)}
							scrolling={scrolling}
							onChange={this.onTimeChange}
							dropdownSelectedClass={styles['selected-option']}
							dropdownOptionsClass={styles.text}
							dropdownParentClass={styles.dropdown}
						/>
					</div>
					}
				</div>


				<div styleName="modal-footer">
					<Button
						bordered
						showLoader={this.props.isReturnScheduleLoading}
						onClick={this.scheduleReturn}
						disabled={this.disableButton()}
					>
						Schedule Return
					</Button>
					<Button buttonType="plain" onClick={closeModal} inverted >Cancel</Button>
				</div>
			</Modal>
		);
	}
}

export default CSSModules(ReturnVehicleModal, styles);
