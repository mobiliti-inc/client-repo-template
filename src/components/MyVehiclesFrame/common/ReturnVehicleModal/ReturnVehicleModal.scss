@import '~variables';

.title {
  color: $darker-blue;
  font-family: $futura;
  font-size: 28px;
  font-weight: bold;
  line-height: 31px;
  margin-bottom: 0;
  margin-top: 10px;
  width: 100%;
}

.sub-title {
  color: $darker-blue;
  font-family: $hanken-sans;
  font-size: 16px;
  line-height: 19px;
  margin: 0;
  padding-top: 5px;
  width: 100%;
  font-weight: normal;
}

.subscription-end-date {
  border-bottom: 1px solid rgba(29, 49, 85, 0.2);
  p:first-child {
    font-family: $futura;
    font-size: 18px;
    color: $darker-blue;
    font-weight: 500;
    margin-bottom: 0;
  }

  p:last-child {
    font-family: $hanken-sans;
    font-size: 16px;
    color: $darker-blue;
    margin-top: 3px;
    padding-bottom: 15px;
  }
  
}
.drop-off-date-title {
  font-family: $futura;
  font-size: 18px;
  color: $darker-blue;
  font-weight: 500;
  padding-top: 10px;
}

.date-selection-ctn {
  display: flex;
  justify-content: space-between;
  flex-direction: row;

  @media (max-width: $mobile-width) {
    flex-direction: column;
  }

  .date-picker-ctn {
    display: flex;
    justify-content: center;
    align-items: center;
    > img {
      padding-right: 15px;
    }

    @media (max-width: $mobile-width) {
      div {
        width: 100%;
      }
    }
  }

  .time-drop-down-ctn {
    display: flex;
    justify-content: center;
    align-items: center;
    > img {
      padding-right: 15px;
    }

    @media (max-width: $mobile-width) {
      div {
        width: 100%;
      }
    }
  }

  .selected-option {
    border-bottom: 1px solid rgba(29, 49, 85, 0.2);
    color: $darker-blue;
    font-size: 18px !important;
    padding-bottom: 10px;
    padding-top: 17px;
    color: black;
    font-weight: 400;
    font-family: $hanken-sans;
    display: grid;
    grid-template-columns: 90% 10%;
    align-items: center;
    justify-content: center;

    @media (max-width: $mobile-width) {
      height: 50px;
      margin: 0;
    }
  }

  .text {
    font-size: 16px;
  }

  .header {
    font-size: 16px;
    color: #7B8DA8;
    font-weight: 500;
  }
}
.modal-footer {
  padding-top: 10%;
  display: flex;
  justify-content: space-between;
}

.dropdown {
  width: 300px;
}

.modal {
  @media (min-width: $mobile-width) {
    width: 100%;
  }

  @media (min-width: $tablet-width) {
    width: 40%;
  }
}
