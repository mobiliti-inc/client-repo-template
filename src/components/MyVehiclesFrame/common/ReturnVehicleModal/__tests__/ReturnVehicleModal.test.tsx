import ReturnVehicleModal from '../ReturnVehicleModal';

describe('ReturnVehicleModal', () => {
	const props = {
		isReturnScheduleLoading: false,
		onSelectDropOffDay: jest.fn(),
		onSelectDropOffTime: jest.fn(),
		onScheduleVehicleReturn: jest.fn(),
		selectedDropOffDay: '03/03/2019',
		selectedDropOffTime: '11:00AM - 1:00PM',
		subscriptionID: '45',
		closeModal: jest.fn(),
		scrolling: false,
		subscriptionEndDate: '02/28/2019',
		modalVisible: true,
		dealerOpenDays: [
			{
				open_day: '2/23/2019',
				hours: [
					'10:00 AM - 12:00 PM',
					'12:00 PM - 2:00 PM',
					'2:00 PM - 4:00 PM',
					'4:00 PM - 6:00 PM',
					'6:00 PM - 7:00 PM'
				]
			}
		],
	};

	const component = <ReturnVehicleModal {...props} />;
	it('renders', () => {
		const tree = shallow(component);
		expect(tree).toMatchSnapshot();
	});
});
