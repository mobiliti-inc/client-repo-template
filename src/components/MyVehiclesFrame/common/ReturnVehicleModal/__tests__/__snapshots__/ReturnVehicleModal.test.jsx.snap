// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReturnVehicleModal renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReturnVehicleModal
    closeModal={[MockFunction]}
    dealerOpenDays={
      Array [
        Object {
          "hours": Array [
            "10:00 AM - 12:00 PM",
            "12:00 PM - 2:00 PM",
            "2:00 PM - 4:00 PM",
            "4:00 PM - 6:00 PM",
            "6:00 PM - 7:00 PM",
          ],
          "open_day": "2/23/2019",
        },
      ]
    }
    isReturnScheduleLoading={false}
    modalVisible={true}
    onScheduleVehicleReturn={[MockFunction]}
    onSelectDropOffDay={[MockFunction]}
    onSelectDropOffTime={[MockFunction]}
    scrolling={false}
    selectedDropOffDay="03/03/2019"
    selectedDropOffTime="11:00AM - 1:00PM"
    subscriptionEndDate="02/28/2019"
    subscriptionID="45"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bodyClassName": "",
      "buttonClassName": null,
      "buttonText": "",
      "children": Array [
        <p
          className="title"
        >
          Schedule Vehicle Return
        </p>,
        <p
          className="sub-title"
        >
          If you are returning your vehicle after one month of use you could receive a credit based on your vehicle usage for the current month.
        </p>,
        <div
          className="subscription-end-date"
        >
          <p>
            Thursday, February 28th
          </p>
          <p>
            Subscription End Date
          </p>
        </div>,
        <p
          className="drop-off-date-title"
        >
          Drop off Date/Time
        </p>,
        <div
          className="date-selection-ctn"
        >
          <div
            className="date-picker-ctn"
          >
            <img
              alt="calendar-icon"
              src="test-file-stub"
            />
            <CustomDatePicker
              errorMessage={null}
              id=""
              isDayBlocked={[Function]}
              isOutsideRange={[Function]}
              isValid={true}
              label=""
              license={false}
              negative={false}
              numberOfYears={2}
              onDatePicked={[Function]}
              openToDate={null}
              placeholder="Date"
              selectedDate={null}
            />
          </div>
        </div>,
        <div
          className="modal-footer"
        >
          <Button
            bordered={true}
            buttonType="standard"
            className=""
            customStyles=""
            disabled={true}
            icon={null}
            inverted={false}
            onClick={[Function]}
            ripple={false}
            showLoader={false}
          >
            Schedule Return
          </Button>
          <Button
            bordered={false}
            buttonType="plain"
            className=""
            customStyles=""
            disabled={false}
            icon={null}
            inverted={true}
            onClick={[MockFunction]}
            ripple={false}
            showLoader={false}
          >
            Cancel
          </Button>
        </div>,
      ],
      "footerClassName": "",
      "freezeOverlay": false,
      "headerClassName": "",
      "icon": <div />,
      "modalParentClass": "modal",
      "noIcon": false,
      "noPointer": false,
      "onButtonClick": [Function],
      "onIconClose": [MockFunction],
      "overlay": "dark-gray",
      "plain": true,
      "plainHeader": false,
      "scrolling": false,
      "showButtonLoader": false,
      "showCloseIcon": false,
      "subTitle": "",
      "title": "",
      "visible": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Schedule Vehicle Return",
          "className": "title",
        },
        "ref": null,
        "rendered": "Schedule Vehicle Return",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "If you are returning your vehicle after one month of use you could receive a credit based on your vehicle usage for the current month.",
          "className": "sub-title",
        },
        "ref": null,
        "rendered": "If you are returning your vehicle after one month of use you could receive a credit based on your vehicle usage for the current month.",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              Thursday, February 28th
            </p>,
            <p>
              Subscription End Date
            </p>,
          ],
          "className": "subscription-end-date",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Thursday, February 28th",
            },
            "ref": null,
            "rendered": "Thursday, February 28th",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Subscription End Date",
            },
            "ref": null,
            "rendered": "Subscription End Date",
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Drop off Date/Time",
          "className": "drop-off-date-title",
        },
        "ref": null,
        "rendered": "Drop off Date/Time",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="date-picker-ctn"
            >
              <img
                alt="calendar-icon"
                src="test-file-stub"
              />
              <CustomDatePicker
                errorMessage={null}
                id=""
                isDayBlocked={[Function]}
                isOutsideRange={[Function]}
                isValid={true}
                label=""
                license={false}
                negative={false}
                numberOfYears={2}
                onDatePicked={[Function]}
                openToDate={null}
                placeholder="Date"
                selectedDate={null}
              />
            </div>,
            false,
          ],
          "className": "date-selection-ctn",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt="calendar-icon"
                  src="test-file-stub"
                />,
                <CustomDatePicker
                  errorMessage={null}
                  id=""
                  isDayBlocked={[Function]}
                  isOutsideRange={[Function]}
                  isValid={true}
                  label=""
                  license={false}
                  negative={false}
                  numberOfYears={2}
                  onDatePicked={[Function]}
                  openToDate={null}
                  placeholder="Date"
                  selectedDate={null}
                />,
              ],
              "className": "date-picker-ctn",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "calendar-icon",
                  "src": "test-file-stub",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "errorMessage": null,
                  "id": "",
                  "isDayBlocked": [Function],
                  "isOutsideRange": [Function],
                  "isValid": true,
                  "label": "",
                  "license": false,
                  "negative": false,
                  "numberOfYears": 2,
                  "onDatePicked": [Function],
                  "openToDate": null,
                  "placeholder": "Date",
                  "selectedDate": null,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          false,
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Button
              bordered={true}
              buttonType="standard"
              className=""
              customStyles=""
              disabled={true}
              icon={null}
              inverted={false}
              onClick={[Function]}
              ripple={false}
              showLoader={false}
            >
              Schedule Return
            </Button>,
            <Button
              bordered={false}
              buttonType="plain"
              className=""
              customStyles=""
              disabled={false}
              icon={null}
              inverted={true}
              onClick={[MockFunction]}
              ripple={false}
              showLoader={false}
            >
              Cancel
            </Button>,
          ],
          "className": "modal-footer",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bordered": true,
              "buttonType": "standard",
              "children": "Schedule Return",
              "className": "",
              "customStyles": "",
              "disabled": true,
              "icon": null,
              "inverted": false,
              "onClick": [Function],
              "ripple": false,
              "showLoader": false,
            },
            "ref": null,
            "rendered": "Schedule Return",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bordered": false,
              "buttonType": "plain",
              "children": "Cancel",
              "className": "",
              "customStyles": "",
              "disabled": false,
              "icon": null,
              "inverted": true,
              "onClick": [MockFunction],
              "ripple": false,
              "showLoader": false,
            },
            "ref": null,
            "rendered": "Cancel",
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bodyClassName": "",
        "buttonClassName": null,
        "buttonText": "",
        "children": Array [
          <p
            className="title"
          >
            Schedule Vehicle Return
          </p>,
          <p
            className="sub-title"
          >
            If you are returning your vehicle after one month of use you could receive a credit based on your vehicle usage for the current month.
          </p>,
          <div
            className="subscription-end-date"
          >
            <p>
              Thursday, February 28th
            </p>
            <p>
              Subscription End Date
            </p>
          </div>,
          <p
            className="drop-off-date-title"
          >
            Drop off Date/Time
          </p>,
          <div
            className="date-selection-ctn"
          >
            <div
              className="date-picker-ctn"
            >
              <img
                alt="calendar-icon"
                src="test-file-stub"
              />
              <CustomDatePicker
                errorMessage={null}
                id=""
                isDayBlocked={[Function]}
                isOutsideRange={[Function]}
                isValid={true}
                label=""
                license={false}
                negative={false}
                numberOfYears={2}
                onDatePicked={[Function]}
                openToDate={null}
                placeholder="Date"
                selectedDate={null}
              />
            </div>
          </div>,
          <div
            className="modal-footer"
          >
            <Button
              bordered={true}
              buttonType="standard"
              className=""
              customStyles=""
              disabled={true}
              icon={null}
              inverted={false}
              onClick={[Function]}
              ripple={false}
              showLoader={false}
            >
              Schedule Return
            </Button>
            <Button
              bordered={false}
              buttonType="plain"
              className=""
              customStyles=""
              disabled={false}
              icon={null}
              inverted={true}
              onClick={[MockFunction]}
              ripple={false}
              showLoader={false}
            >
              Cancel
            </Button>
          </div>,
        ],
        "footerClassName": "",
        "freezeOverlay": false,
        "headerClassName": "",
        "icon": <div />,
        "modalParentClass": "modal",
        "noIcon": false,
        "noPointer": false,
        "onButtonClick": [Function],
        "onIconClose": [MockFunction],
        "overlay": "dark-gray",
        "plain": true,
        "plainHeader": false,
        "scrolling": false,
        "showButtonLoader": false,
        "showCloseIcon": false,
        "subTitle": "",
        "title": "",
        "visible": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Schedule Vehicle Return",
            "className": "title",
          },
          "ref": null,
          "rendered": "Schedule Vehicle Return",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "If you are returning your vehicle after one month of use you could receive a credit based on your vehicle usage for the current month.",
            "className": "sub-title",
          },
          "ref": null,
          "rendered": "If you are returning your vehicle after one month of use you could receive a credit based on your vehicle usage for the current month.",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                Thursday, February 28th
              </p>,
              <p>
                Subscription End Date
              </p>,
            ],
            "className": "subscription-end-date",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Thursday, February 28th",
              },
              "ref": null,
              "rendered": "Thursday, February 28th",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Subscription End Date",
              },
              "ref": null,
              "rendered": "Subscription End Date",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Drop off Date/Time",
            "className": "drop-off-date-title",
          },
          "ref": null,
          "rendered": "Drop off Date/Time",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="date-picker-ctn"
              >
                <img
                  alt="calendar-icon"
                  src="test-file-stub"
                />
                <CustomDatePicker
                  errorMessage={null}
                  id=""
                  isDayBlocked={[Function]}
                  isOutsideRange={[Function]}
                  isValid={true}
                  label=""
                  license={false}
                  negative={false}
                  numberOfYears={2}
                  onDatePicked={[Function]}
                  openToDate={null}
                  placeholder="Date"
                  selectedDate={null}
                />
              </div>,
              false,
            ],
            "className": "date-selection-ctn",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <img
                    alt="calendar-icon"
                    src="test-file-stub"
                  />,
                  <CustomDatePicker
                    errorMessage={null}
                    id=""
                    isDayBlocked={[Function]}
                    isOutsideRange={[Function]}
                    isValid={true}
                    label=""
                    license={false}
                    negative={false}
                    numberOfYears={2}
                    onDatePicked={[Function]}
                    openToDate={null}
                    placeholder="Date"
                    selectedDate={null}
                  />,
                ],
                "className": "date-picker-ctn",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "calendar-icon",
                    "src": "test-file-stub",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "errorMessage": null,
                    "id": "",
                    "isDayBlocked": [Function],
                    "isOutsideRange": [Function],
                    "isValid": true,
                    "label": "",
                    "license": false,
                    "negative": false,
                    "numberOfYears": 2,
                    "onDatePicked": [Function],
                    "openToDate": null,
                    "placeholder": "Date",
                    "selectedDate": null,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            false,
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Button
                bordered={true}
                buttonType="standard"
                className=""
                customStyles=""
                disabled={true}
                icon={null}
                inverted={false}
                onClick={[Function]}
                ripple={false}
                showLoader={false}
              >
                Schedule Return
              </Button>,
              <Button
                bordered={false}
                buttonType="plain"
                className=""
                customStyles=""
                disabled={false}
                icon={null}
                inverted={true}
                onClick={[MockFunction]}
                ripple={false}
                showLoader={false}
              >
                Cancel
              </Button>,
            ],
            "className": "modal-footer",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bordered": true,
                "buttonType": "standard",
                "children": "Schedule Return",
                "className": "",
                "customStyles": "",
                "disabled": true,
                "icon": null,
                "inverted": false,
                "onClick": [Function],
                "ripple": false,
                "showLoader": false,
              },
              "ref": null,
              "rendered": "Schedule Return",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bordered": false,
                "buttonType": "plain",
                "children": "Cancel",
                "className": "",
                "customStyles": "",
                "disabled": false,
                "icon": null,
                "inverted": true,
                "onClick": [MockFunction],
                "ripple": false,
                "showLoader": false,
              },
              "ref": null,
              "rendered": "Cancel",
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
